app/controllers/ems_infra_controller.rb

```diff
@@ -111,6 +111,48 @@ def scaledown
     end
   end
 
+  def ironic_register
+    assert_privileges("host_ironic_register_nodes")
+    redirect_to ems_infra_path(params[:id], :display => "hosts") if params[:cancel]
+
+    # Hiding the toolbars
+    @in_a_form = true
+    drop_breadcrumb(:name => _("Register Ironic Nodes"), :url => "/ems_infra/ironic_register")
+
+    @infra = ManageIQ::Providers::Openstack::InfraManager.find(params[:id])
+
+    if params[:register]
+      uploaded_file = params[:nodes_json_file]
+      if uploaded_file.nil?
+        log_and_flash_message(_("Please select a JSON file containing the nodes you would like to register."))
+        return
+      end
+
+      begin
+        nodes_json = parse_json(uploaded_file)
+        if nodes_json.nil?
+          log_and_flash_message(_("JSON file format is incorrect, missing 'nodes'."))
+        end
+      rescue => ex
+        log_and_flash_message(_("Cannot parse JSON file: %{message}") %
+                                  {:message => ex})
+      end
+
+      if nodes_json
+        state, workflow_execution_id = register_nodes(nodes_json)
```

Comment:
I talked to @rwsu about this; it takes a few seconds at most, so this won't really block.  Introspection will take longer, but that's not part of this patch, and that will be implemented as a non-blocking task.